umbraco/Umbraco-CMS
# V8: Improve plugin language handling

### Prerequisites


If there's an existing issue for this PR then this fixes: https://github.com/umbraco/Umbraco-CMS/issues/4371

### Description

This PR allows for more flexible plugin language files as discussed on #4371. 

I have removed all restrictions/filtering on file name length as `LocalizedTextServiceFileSources` already handles validation of the file content and mapping to the correct culture.

### Testing this PR

Provided that your Umbraco user is set up to en-US localization:

1. Add this plugin to `App_Plugins`: [LanguageTestPlugin.zip](https://github.com/umbraco/Umbraco-CMS/files/2837096/LanguageTestPlugin.zip)
2. Replace the contents of `~/Config/Lang/en-US.user.xml` with this:
```xml
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<language>
    <area alias="treeHeaders">
        <key alias="content">User defined label here</key>
    </area>
</language>
```
3. Restart the site.
4. Verify that the test dashboard shows the en-US labels defined in `~/App_Plugins/LanguageTestPlugin/Lang/en-US.xml` and that the content tree header is updated according to `~/Config/Lang/en-US.user.xml`:
![image](https://user-images.githubusercontent.com/7405322/52354369-92ba7200-2a30-11e9-9a97-a79d19ce8dfc.png)
5. Rename `~/App_Plugins/LanguageTestPlugin/Lang/en-US.xml` to `~/App_Plugins/LanguageTestPlugin/Lang/en_us.xml` 
6. Restart the site.
7. Verify that the test dashboard still shows the en-US labels.
8. Delete `~/App_Plugins/LanguageTestPlugin/Lang/en-US.xml` and rename `~/App_Plugins/LanguageTestPlugin/Lang/xx.xml` to `~/App_Plugins/LanguageTestPlugin/Lang/en.xml`.
9. Restart the site.
10. Verify that the test dashboard now shows the en labels defined in `~/App_Plugins/LanguageTestPlugin/Lang/en.xml`:
![image](https://user-images.githubusercontent.com/7405322/52354586-ffce0780-2a30-11e9-98d8-9033daf31cdb.png)
